### PR TITLE
Resolve Slack rate limit error by adjusting how we render channels on the Slack config page

### PR DESCRIPTION
**Pull request by lilbitner**
## Purpose

This PR resolves the rate limit issue encountered by Stripe when fetching all slack channels on the Slack app configuration page. 

## Approach

Through lots of trial and error and eventually finding a useful endpoint, I ended up changing the way the notifications are configured with respect to what slack channels are available to configure. Instead of listing all slack channels (the endpoint that can cause rate limit headaches when returning thousands of channels), I instead list the channels that have the Contentful app integrated (which is a requirement for the integration itself). Please refer to [this](https://slack.com/openid/connect/login_initiate_redirect?login_hint=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwczpcL1wvc2xhY2suY29tIiwic3ViIjoibGlsbGlhbi5iaXRuZXJAY29udGVudGZ1bC5jb20iLCJhdWQiOiI5MjY1OTQ1MDAzNy4zMjIwNjE1ODI4ODYiLCJleHAiOjE2OTE2OTEyNjIsImlhdCI6MTY5MTY5MTI1NSwiYXV0aF90aW1lIjoxNjkxNjkxMjU1LCJodHRwczpcL1wvc2xhY2suY29tXC91c2VyX2lkIjoiVTA0TU1VTTBDSEoiLCJodHRwczpcL1wvc2xhY2suY29tXC90ZWFtX2lkIjoiVDAyOTE3UjQ2IiwiaHR0cHM6XC9cL3NsYWNrLmNvbVwvdGFyZ2V0X3VyaSI6Imh0dHBzOlwvXC93d3cubG9vbS5jb21cL3NoYXJlXC83MDlkNDk4MTdkMDM0ODA2YWM5ZTlkY2ViNzc0MjczMiJ9.KzmaSosX6xP8iZ9Z9ibye8u6v97bSDYlxMgADf3HuR20GAB2wS4_gGgK7SqLRxDLU3UeZ0fDXDzcgjHd07ydwvILodaLM5-b07Y5voW2gTsdjjkbzEnpMiM1jFTK254PENkUbfgZ3pnYTrAWQwqE3N85tWI3VhJAwn5Fv6H8hvnJygmGpbIYOtljf509nXWU5Xqh9RRS-7N1ctOmXi96_h2S0evfnvgPPYTKtXcQyJsqw6m4PyDTKwcP7XoydbHeGnC98KqwodOdEkGt2eQBBMmBk8mNdJWaC--wXNwHlxp3Ww1CNPl4W905d80A_uplSj5zSH6G9OxfOmK0JsH8iQ) video (which most of you have seen) for more details. 

Below you will find the various iterations of the new flow, and copy suggestions very welcome!! 

### Need to select a channel: 
Screenshot 2023-08-10 at 11 13 59 AM

### Need to change a channel: 
Screenshot 2023-08-10 at 11 17 54 AM

### Error state when fetching channel: 
Screenshot 2023-08-10 at 12 10 04 PM

### Rendering channels within modal: 
Screenshot 2023-08-10 at 12 11 29 PM

### Selecting and applying channel from modal: 
Screenshot 2023-08-10 at 1 11 55 PM

### Warning when no channels are available to list: 
Screenshot 2023-08-10 at 11 28 01 AM

### Error when channels cannot be fetched: 
Screenshot 2023-08-10 at 11 23 41 AM

## Testing steps

Play around with the Slack app configuration locally :) Please let me know if you have difficulty with this!

## Breaking Changes

None that I am aware of. The changes should be backwards compatible with already configured applications. 

## Note

I am still looking into some of the already weird nuances of how the Lambda is set-up with errors. No issues related to changes in this PR but seeing if I can ensure that server errors are properly thrown (which right now they aren't really). Looking into it as this PR is reviewed, just to expedite these changes :)

### CODE REVIEW

File: apps/slack/frontend/src/components/ChannelModal/ChannelListModal.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        </>
+      );
+    }
+
+    if (channels && !channels.length) {
+      return (
+        <Note title="No channels available">
+          There are currently no channels within your Slack workspace where the Contentful app has been added. 
+          Make sure to add the
+          {renderContentfulAppLink()} app to the Slack channels that you would like listed here.
```
Comment (gavinmatt) [inline apps/slack/frontend/src/components/ChannelModal/ChannelListModal.tsx:91]: Maybe add "Only channels with the {renderContentfulAppLink()} app present will show in the channel list" at the end?

Comment (lilbitner) [reply apps/slack/frontend/src/components/ChannelModal/ChannelListModal.tsx:91]: yeah this is a great call, will adjust!

File: apps/slack/frontend/src/components/NotificationItem/NotificationItem.tsx [hunk 2]
```diff
@@ -124,32 +168,19 @@ export const NotificationItem = ({
...
-        </FormControl>
-        <IconButton
-          icon={<DeleteIcon variant="secondary" />}
-          aria-label="Delete notification"
-          onClick={() => removeNotificationAtIndex(index)}
-          variant="transparent"
-          className={styles.delete}
-        />
+          <FormControl className={styles.select}>
+          <FormControl.Label>Selected slack channel</FormControl.Label>
```
Comment (whitelisab) [inline apps/slack/frontend/src/components/NotificationItem/NotificationItem.tsx:172]: nit: Capitalize `Slack` here

**Review by whitelisab — COMMENTED** [review]
Overall I think this works really well and the code is very clean! 🎊  I tested adding the Contentful app to a new channel and I didn't see it reflected in the list though. Not sure if I missed something, but we would want to make sure that works properly.

**Comment by lilbitner** [comment]
> Overall I think this works really well and the code is very clean! 🎊 I tested adding the Contentful app to a new channel and I didn't see it reflected in the list though. Not sure if I missed something, but we would want to make sure that works properly.

Ah yeah locally you would have to add the Contentful app AND Team Integrations app because this Team Integrations app helps facilitate local development (sorry that I didn't make that clearer). I tested in production via the Slack API that just adding the Contentful app returns the correct channels! Thank you for testing this!!! @whitelisab

**Review by whitelisab — APPROVED** [review]
Awesome, thanks for clarifying how to add the app to channels locally. Looks good to me! ⭐

**Pull request merged**